src/model/loss.py

```diff
@@ -23,11 +24,24 @@ def forward(self, hr_fake, hr_real, d_fake, d_real):
         """
 
         # Using straight MSE for now
-        g_loss = self.GLoss(hr_fake, hr_real, d_fake, content_choice='feat')
-        d_loss = self.DLoss(d_fake, d_real)
+        # g_loss = self.GLoss(hr_fake, hr_real, d_fake, content_choice='feat')
+        # d_loss = self.DLoss(d_fake, d_real)
+        g_loss = self.GLoss_new(hr_fake, hr_real, d_fake)
+        d_loss = self.DLoss_new(d_fake, d_real)
 
         return d_loss, g_loss
 
+    def GLoss_new(self, generated_image, target, d_fake):
+        content_loss = self.mse(generated_image, target)
+
+        adverserial_loss = torch.mean(-F.logsigmoid(d_fake))
+        #adverserial_loss = torch.mean(torch.ones_like(d_fake) - d_fake)
+
+        return content_loss + 0.001 * adverserial_loss
+    
+    def DLoss_new(self, d_fake, d_real):
+        return 1 - d_real.mean() + d_fake.mean()
```

Comment:
Can you explain how this works? We get normalized probabilities out of the model (after the sigmoid), how does taking the average of that relate to binary cross entropy?

Reply:
Your original DLoss quickly descent to 0 after just 1 batch of data. I was not sure if it was correct at the time, so I wrote a new DLoss. They seem to perform similarly if I remember correctly. The intuition behind this is that GLoss is a float number and we should have a loss 1-(the difference between real and fake).